compiler/src/scala/quoted/runtime/impl/QuoteMatcher.scala

```diff
@@ -449,6 +474,26 @@ object QuoteMatcher {
      */
     case OpenTree(tree: Tree, patternTpe: Type, args: List[Tree], env: Env)
 
+    /** The Definitions object */
+    def defn(using Context): Definitions = ctx.definitions
+
+    /*
+     * PR-17567 Remaining TODOs
+     * * [ ] Implicit / Contextual parameters
+     * * [x] Nested Method Types
+     * * [ ] Erased Types
+     */
+    def adaptTypes(tpe: Type)(using Context): Type =
+      new Types.TypeMap {
+          def apply(tp: Types.Type): Types.Type = tp match
+            case tp: MethodType => {
+              val paramInfos = tp.paramInfos map adaptTypes
+              val resultType = adaptTypes(tp.resultType)
+              return defn.FunctionOf(paramInfos, resultType)
+            }
+            case _ => mapOver(tp)
```

Comment:
We do not want to do the recursion on all nested types. We only need to map the outermost `MethodType` (and `PolyType` in later on) into lambda types.

Note that there are cases where we can encounter `MethodType` that we do not want to if we have refinement types. For example 
```scala
case '{
  def f(x: { def g(i: Int): Int } ): Unit = ...
  $h(f)
}
```
where `h` would have type `Expr[({ def g(i: Int): Int }) => Unit]` where the `MethodType` of `g` would not be modified.

Reply:
This is probably what we need
```scala
    def adaptTypes(tpe: Type)(using Context): Type = tpe match
      case tp: MethodType =>
        defn.FunctionOf(tp.paramInfos, adaptTypes(tp.resultType)/*, tp.isImplicitMethod*/) // or tp.isContextualMethod
      case _ =>
        tpe
```

Reply:
Fixed, thank you. I should add this test case.